include/seqan3/range/view/pairwise_combine.hpp

```diff
@@ -0,0 +1,690 @@
+         * other entries are empty. Using this scheme one can use the properties of triangular numbers, where the
+         * diagonal index of this matrix can be computed using triangular roots
+         * (see https://stackoverflow.com/questions/27086195/linear-index-upper-triangular-matrix).
+         * Given these properties, one can compute the matrix index (i, j) from the linearised matrix index and vice
+         * versa.
+         */
+        constexpr size_t to_index() const
+            noexcept(noexcept(std::declval<underlying_iterator_type &>() - std::declval<underlying_iterator_type &>()))
+        //!\cond
+            requires std::RandomAccessIterator<underlying_iterator_type>
+        //!\endcond
+        {
+            size_t src_size = end_it - begin_it;
+            size_t index_i = first_it - begin_it;
+            size_t index_j = second_it - begin_it;
+            return (src_size * (src_size - 1)/2) - (src_size - index_i) * ((src_size - index_i) - 1)/2 +
+                   index_j - index_i - 1;
+        }
+
+        /*!\brief Sets the iterator to the given index.
+         * \param[in] index The index to set the iterator to.
+         * \copydetails to_index()
+         */
+        constexpr void from_index(size_t const index)
+            noexcept(noexcept(std::declval<underlying_iterator_type &>() - std::declval<underlying_iterator_type &>()) &&
+                     noexcept(std::declval<underlying_iterator_type &>() + 1))
+        //!\cond
+            requires std::RandomAccessIterator<underlying_iterator_type>
+        //!\endcond
+        {
+            size_t src_size = end_it - begin_it;
+            size_t index_i = src_size - 2 -
+                             std::floor(std::sqrt(-8 * index + 4 * src_size * (src_size - 1) - 7)/2.0 - 0.5);
+            size_t index_j = index + index_i + 1 - src_size * (src_size - 1)/2 + (src_size - index_i) *
+                             ((src_size - index_i) - 1)/2;
+            first_it = begin_it + index_i;
+            second_it = begin_it + index_j;
+        }
+
+        //!\brief The iterator pointing to the first element of the pairwise combination.
+        underlying_iterator_type first_it{};
+        //!\brief The iterator pointing to the second element of the pairwise combination.
+        underlying_iterator_type second_it{};
+        //!\brief The begin of the underlying range.
+        underlying_iterator_type begin_it{};
+        //!\brief The end of the underlying range.
+        underlying_iterator_type end_it{};
+    };
+
+public:
+
+    //!\name Member types
+    //!\{
+    //!\brief The iterator type.
+    using iterator = iterator_type<underlying_range_type>;
+    //!\brief The const iterator type. Evaluates to void if the underlying range is not const iterable.
+    using const_iterator = transformation_trait_or_t<std::type_identity<iterator_type<underlying_range_type const>>,
+                                                     void>;
+    //!\}
```

Comment:
what about the other associated types, like `value_type`, `reference` ...? We provide those for all our views and containers (although not strictly required by range concepts).

Reply:
I agree, that we should keep it consistent. And I propose removing them as they can introduce subtle errors and it is not necessary as in most cases you access them via the meta-functions, through the iterator interface.

Reply:
> I agree, that we should keep it consistent.

Ok, then please add them for now.

> And I propose removing them as they can introduce subtle errors and it is not necessary as in most cases you access them via the meta-functions, through the iterator interface.

Let's discuss it at the next developer meeting. I don't have strong feelings about it, although I have a slight preference for keeping them, especially since our iterators are usually not publicly specified, while the ranges are.

Reply:
I added it already

Reply:
I just saw you already did!